### ISSUE TITLE
invalid aes key size

### ISSUE THREAD

**Issue by jelhan** [bitwiseshiftleft/sjcl#318]
Since update from 1.0.4 to 1.0.6 I'm getting an `invalid aes key size` error in my application.

I'm using `sjcl.encrypt()` with standard parameters, e.g.: `sjcl.encrypt("abcdefghijklmnopqrstuvwxyz012345789", "default title");` I tried to reproduce in a simple example but wasn't able. But while debugging I noticed that the issue is related to compression. If I build with `./configure --compression=none` everything works fine. When building with compression I'm getting 3 warnings, not quite sure if it's related:

```
$ make
cat core/sjcl.js core/aes.js core/bitArray.js core/codecString.js core/codecHex.js core/codecBase32.js core/codecBase64.js core/sha256.js core/ccm.js core/ocb2.js core/gcm.js core/hmac.js core/pbkdf2.js core/random.js core/convenience.js core/exports.js  > core.js
compress/compress_with_closure.sh core.js > core_closure.js
._tmpRC.js:469: WARNING - optional arguments must be at the end
  _shiftRight: function (a, shift, carry, out) {
               ^

._tmpRC.js:503: WARNING - parameter 'bitArray' does not appear in <anonymous>'s parameter list
  'byteswapM': function(a) {
               ^

._tmpRC.js:1639: WARNING - optional arguments must be at the end
sjcl['misc']['pbkdf2'] = function (password, salt, count, length, Prff) {
                         ^

0 error(s), 3 warning(s)
cp core_closure.js sjcl.js
```

**Comment by jelhan** [comment]
I got this fixed by not addressing `sjcl` as a global.

**Comment by Nilos** [comment]
This is most likely due to a shared global variable by the compressed sjcl code and your code. I hoped that the closure compiler would fix all of those issues and add all of its variables into a private namespace but maybe it does not. Will investigate as soon as I find time!

**Comment by jelhan** [comment]
Yeah seems so. I [could fix it](https://github.com/jelhan/croodle/commit/ae0ad7bd42102955ec2b133cb82311895d70f1d3) by importing sjcl as a module and not accessing it as a global.

**Comment by mellogarrett** [comment]
I'm getting this error as well. I've tried 128, 192, and 256 bit keys. 

How do I fix?
